packages/fx-core/src/plugins/resource/bot/v3/index.ts

```diff
@@ -0,0 +1,521 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { hooks } from "@feathersjs/hooks";
+import {
+  AzureAccountProvider,
+  AzureSolutionSettings,
+  EnvConfig,
+  err,
+  FxError,
+  Inputs,
+  ok,
+  Result,
+  TokenProvider,
+  v2,
+  v3,
+  Void,
+} from "@microsoft/teamsfx-api";
+import * as path from "path";
+import { Service } from "typedi";
+import { ArmTemplateResult } from "../../../../common/armInterface";
+import { Bicep, ConstantString } from "../../../../common/constants";
+import {
+  generateBicepFromFile,
+  getResourceGroupNameFromResourceId,
+  getSiteNameFromResourceId,
+  getSubscriptionIdFromResourceId,
+} from "../../../../common/tools";
+import { CommonErrorHandlerMW } from "../../../../core/middleware/CommonErrorHandlerMW";
+import { getTemplatesFolder } from "../../../../folder";
+import {
+  AzureSolutionQuestionNames,
+  BotOptionItem,
+  MessageExtensionItem,
+} from "../../../solution/fx-solution/question";
+import { BuiltInFeaturePluginNames } from "../../../solution/fx-solution/v3/constants";
+import {
+  AzureConstants,
+  BotBicep,
+  DeployConfigs,
+  FolderNames,
+  MaxLengths,
+  PathInfo,
+  ProgressBarConstants,
+  TemplateProjectsConstants,
+} from "../constants";
+import { LanguageStrategy } from "../languageStrategy";
+import { ProgressBarFactory } from "../progressBars";
+import { Messages } from "../resources/messages";
+import fs from "fs-extra";
+import { CommonStrings, ConfigNames, PluginLocalDebug } from "../resources/strings";
+import { TokenCredentialsBase } from "@azure/ms-rest-nodeauth";
+import * as factory from "../clientFactory";
+import { ResourceNameFactory } from "../utils/resourceNameFactory";
+import { AADRegistration } from "../aadRegistration";
+import { IBotRegistration } from "../appStudio/interfaces/IBotRegistration";
+import { AppStudio } from "../appStudio/appStudio";
+import { DeployMgr } from "../deployMgr";
+import * as utils from "../utils/common";
+import * as appService from "@azure/arm-appservice";
+import { AzureOperations } from "../azureOps";
+import { getZipDeployEndpoint } from "../utils/zipDeploy";
+import {
+  ScaffoldAction,
+  ScaffoldActionName,
+  ScaffoldContext,
+  scaffoldFromTemplates,
+} from "../../../../common/template-utils/templatesActions";
+import { ProgrammingLanguage } from "../enums/programmingLanguage";
+import {
+  CheckThrowSomethingMissing,
+  PackDirectoryExistenceError,
+  PreconditionError,
+  TemplateZipFallbackError,
+  UnzipError,
+} from "./error";
+import { ensureSolutionSettings } from "../../../solution/fx-solution/utils/solutionSettingsHelper";
+
+@Service(BuiltInFeaturePluginNames.bot)
+export class NodeJSBotPluginV3 implements v3.FeaturePlugin {
+  name = BuiltInFeaturePluginNames.bot;
+  displayName = "NodeJS Bot";
+  async pluginDependencies(ctx: v2.Context, inputs: Inputs): Promise<Result<string[], FxError>> {
+    return ok([BuiltInFeaturePluginNames.identity]);
```

Comment:
Mark: any background about this method?

Reply:
Bot plugin depends on identity, if identity is missing, the arm template will have errors.
Declaring the dependency here means: if bot is added, identity plugin should also be added together.